site/lib/mergeHoldings.ts

```diff
@@ -0,0 +1,77 @@
+import { Holding } from "components/pages/Pledge/types";
+import { Dictionary, groupBy } from "lodash";
+
+export const mergeHoldings = (holdings: Holding[][]) => {
+  const groupedHoldings = groupHoldings(holdings);
+  const calculatedChangeHoldings = calculateChange(groupedHoldings);
+  const mergedHoldings = calculateTokenValues(calculatedChangeHoldings);
+  return mergedHoldings;
+};
+
+export const groupHoldings = (holdings: Holding[][]) =>
```

Comment:
can we get some comments on each util for newcomers and dumdums to outline the "why" of all this code

Reply:
good idea, comments added